lib/peer.js

```diff
@@ -222,6 +222,11 @@ export default class Peer extends EventEmitter {
     if (peerId === this.swarm.peerId) {
       return this.destroy(new Error('refusing to connect to ourselves'))
     }
+    // Prevent connecting to the same peer via different transports
+    const alreadyConnected = this.swarm.wires.find(wire => wire.peerId === peerId)
+    if (alreadyConnected) {
+      return this.destroy(new Error(`already connected to peer via ${alreadyConnected.type}`))
+    }
```

Comment:
I'm fairly sure this check existed somewhere already, or something similar

Reply:
There is this [check](https://github.com/webtorrent/webtorrent/blob/a3a6e52609cbd81da584a13fc3b3b85bebfb5db4/lib/torrent.js#L889), but this doesn't work as the `peer.id` for a TCP/uTP connection is the IP:PORT, but the `peer.id` for a WebRTC connection is the peerId.

Reply:
wouldn't it be better then to replace that check with this? // clueless

Reply:
Thinking about it further, yeah, I think we can move the WebRTC check there due to getting the `peerId` from the tracker, which allows us to not waste a peers WebRTC offer or our resources generating an answer to only destroy the peer. 

The existing check is sufficient for uTP/TCP peers due to `IP:PORT` in being unique for a peer regardless of type since uTP/TCP share the same port.

Will make these changes and give them a test to see if the above works.

Reply:
I've removed this part of the PR for now as I've not been able to replicate this again. Will investigate further and open a new issue/PR if I'm able to replicate this again.